resources/lib/utils.py

```diff
@@ -200,3 +200,95 @@ def make_kwparams(params):
     return del_dict_keys(tempparams, ['info', 'type', 'tmdb_id', 'filter_key', 'filter_value',
                                       'with_separator', 'with_id', 'season', 'episode', 'prop_id',
                                       'exclude_key', 'exclude_value'])
+                                      
+
+def recursive_delete_dir(fullpath):
+    '''helper to recursively delete a directory'''
+    success = True
+    if not isinstance(fullpath, unicode):
+        fullpath = fullpath.decode("utf-8")
+    dirs, files = xbmcvfs.listdir(fullpath)
+    for file in files:
+        file = file.decode("utf-8")
+        success = xbmcvfs.delete(os.path.join(fullpath, file))
+    for directory in dirs:
+        directory = directory.decode("utf-8")
+        success = recursive_delete_dir(os.path.join(fullpath, directory))
+    success = xbmcvfs.rmdir(fullpath)
+    return success
+                                      
+                                      
+def _is_url(url):
+    try:  # Python 3
+        from urllib.parse import urlparse
+    except ImportError:  # Python 2
+        from urlparse import urlparse
+
+    try:
+        result = urlparse(url)
+        return all([result.scheme, result.netloc])
+    except ValueError:
+        return False
+        
+
+def _check_url(url, cred):
+    import requests
+
+    if _is_url(url):
+        try:
+            response = requests.head(url, allow_redirects=True, auth=cred)
+            
+            if response.status_code < 300:
+                xbmc.log("URL check passed for {0}: Status code [{1}]".format(url, response.status_code), level=xbmc.LOGDEBUG)
+                return True
+            elif response.status_code < 400:
+                xbmc.log("URL check redirected from {0} to {1}: Status code [{2}]".format(url, response.headers['Location'], response.status_code), level=xbmc.LOGDEBUG)
+                return _check_url(response.headers['Location'])
+            elif response.status_code == 401:
+                xbmc.log("URL requires authentication for {0}: Status code [{1}]".format(url, response.status_code), level=xbmc.LOGDEBUG)
+                return 'auth'
+            else:
+                xbmc.log("URL check failed for {0}: Status code [{1}]".format(url, response.status_code), level=xbmc.LOGDEBUG)
+                return False
+        except Exception as e:
+            xbmc.log("URL check error for {0}: [{1}]".format(url, e), level=xbmc.LOGDEBUG)
+            return False
+    else:
+        xbmc.log("URL is not of a valid schema: {0}".format(url), level=xbmc.LOGDEBUG)
+        return False
+        
+
+def open_url(url, stream=False, check=False, cred=None, count=0):
+    import requests
+
+    if not url:
+        return False
+
+    count = 0
```

Comment:
This line defeats the purpose of doing a retry count because you reset it to zero every time.